python/activity_stream/activity_stream.py

```diff
@@ -968,8 +973,22 @@ def _on_entity_created(self, entity):
         if self.notes_are_selectable and entity["type"] == "Note":
             self._select_on_arrival = entity
         self.entity_created.emit(entity)
-            
-    def _on_reply_clicked(self, note_id):
+
+    def _on_note_content_changed(self, content, note_id):
+        """
+        Callback when a note widget's text `content` field has a new value. This
+        is called once editing has finished, not for each change made during
+        editing.
+
+        :param str content: The Note entity's new `content` field value.
+        :param int note_id: The Note entity id.
+        """
+        sg = self._bundle.shotgun
+        sg.update("Note", note_id, {"content":content})
+        #self._data_manager.rescan() # eldebug
```

Comment:
Done and replaced with an `execute_method` to make the `update` async. However I'm doing `self._data_manager._sg_data_retriever` which is not good. Should I give activity stream its own data retriever or is it ok to expose the one within the data manager?